### PR TITLE
msgpack serializer

### PR DESCRIPTION
**Pull request by liquidiert**
Hey there!
I've seen that there is missing msgpack support and I definitely wanted to have it in :D
So I've copied your existing JSON serializer and changed it to parse msgpack instead.
I also added test cases for serialization, websocket and html transport.
If that is already enough would be nice if you could merge msgpack :)
If not I can add the missing tests before merge.
Greets liquid

### CODE REVIEW

**Comment by konsultaner** [comment]
Thanks @liquidiert for your work. I'm totally willing to merge it. 
Could you please change some snake case variables (i.e. json_serializer) to camelcase -> jsonSerializer.
I'm not sure if msgpack uses binary WebSockets instead of string based WebSockets? I've never implemented msgpack before.

File: lib/src/serializer/msgpack/serializer.dart [hunk 1]
```diff
@@ -0,0 +1,570 @@
...
+      .map((e) => int.tryParse(e)));
+
+  /// Converts a uint8 JSON message into a WAMP message object
+  @override
+  AbstractMessage deserialize(Uint8List msgPack) {
+    return deserializeFromString(msgPack.toString());
+  }
+
+  /// Converts a string JSON message into a WAMP message object
+  AbstractMessage deserializeFromString(String msgPack) {
```
Comment (konsultaner) [inline lib/src/serializer/msgpack/serializer.dart:54]: I don't think that converting from UInt8List to String and back is necessary. You use binary code in your unittest anyway

File: test/serializer/msgpack/serializer_test.dart [hunk 2]
```diff
@@ -0,0 +1,710 @@
...
...8, 116, 115, 195, 170, 115, 117, 98, 115, 99, 114, 105, 98, 101, 114, 129, 168, 102, 101, 97, 116, 117, 114, 101, 115, 133, 174, 99, 97, 108, 108, 95, 99, 97, 110, 99, 101, 108, 105, 110, 103, 194, 172, 99, 97, 108, 108, 95, 116, 105, 109, 101, 111, 117, 116, 194, 180, 112, 97, 121, 108, 111, 97, 100, 95, 116, 114, 97, 110, 115, 112, 97, 114, 101, 110, 99, 121, 195, 184, 112, 114, 111, 103, 114, 101, 115, 115, 105, 118, 101, 95, 99, 97, 108, 108, 95, 114, 101, 115, 117, 108, 116, 115, 194, 183, 115, 117, 98, 115, 99, 114, 105, 112, 116, 105, 111, 110, 95, 114, 101, 118, 111, 99, 97, 116, 105, 111, 110, 195, 169, 112, 117, 98, 108, 105, 115, 104, 101, 114, 129, 168, 102, 101, 97, 116, 117, 114, 101, 115, 132, 184, 112, 117, 98, 108, 105, 115, 104, 101, 114, 95, 105, 100, 101, 110, 116, 105, 102, 105, 99, 97, 116, 105, 111, 110, 195, 189, 115, 117, 98, 115, 99, 114, 105, 98, 101, 114, 95, 98, 108, 97, 99, 107, 119, 104, 105, 116, 101, 95, 108, 105, 115, 116, 105, 110, 103, 195, 179, 112, 117, 98, 108, 105, 115, 104, 101, 114, 95, 101, 120, 99, 108, 117, 115, 105, 111, 110, 195, 180, 112, 97, 121, 108, 111, 97, 100, 95, 116, 114, 97, 110, 115, 112, 97, 114, 101, 110, 99, 121, 195]'));
```
Comment (konsultaner) [inline test/serializer/msgpack/serializer_test.dart:37]: I think you should use UInt8Lists here instead. Where did you get those binary values from?

File: test/transport/websocket/websocket_transport_html_test.dart [hunk 3]
```diff
@@ -14,16 +17,25 @@ void main() {
         'Opening a server connection and simple send receive scenario using a serializer',
         () async {
       // var server = spawnHybridUri('websocket_transport_html_server.dart');
-      var serializer = Serializer();
       // var port = await server.stream.first;
-      var transport = WebSocketTransport(
+      var transportJSON = WebSocketTransport(
           'wss://www.connectanum.com/wamp', // TODO as soon as https://github.com/dart-lang/sdk/issues/40786 is fixed "ws://localhost:$port/wamp",
-          serializer,
+          json_serializer.Serializer(),
```
Comment (konsultaner) [inline test/transport/websocket/websocket_transport_html_test.dart:23]: camel case please 👍

**Review by konsultaner — CHANGES_REQUESTED** [review]

**Comment by liquidiert** [comment]
Sure thing! Regarding the strings I already thought about it but it's kinda tedious due to dartfmt 😬 I think I can push the changes late afternoon

File: lib/src/serializer/msgpack/serializer.dart (hunk 1 above)
Comment (liquidiert) [reply lib/src/serializer/msgpack/serializer.dart:54]: Yeah had the exact same thought 👌

File: test/serializer/msgpack/serializer_test.dart (hunk 2 above)
Comment (liquidiert) [reply test/serializer/msgpack/serializer_test.dart:37]: Will do, good old json to msgpack converter :)

File: test/transport/websocket/websocket_transport_html_test.dart (hunk 3 above)
Comment (liquidiert) [reply test/transport/websocket/websocket_transport_html_test.dart:23]: Kk dartfmt made me :/

File: test/serializer/msgpack/serializer_test.dart (hunk 2 above)
Comment (konsultaner) [reply test/serializer/msgpack/serializer_test.dart:37]: ok 😃

**Comment by konsultaner** [comment]
> Sure thing! Regarding the strings I already thought about it but it's kinda tedious due to dartfmt 😬 I think I can push the changes late afternoon

o'right. awesome. Thank you!

**Review by konsultaner — APPROVED** [review]
looks good, I'll accept the request and check the code for lint and dartfmt errors, update the version and publish it

**Comment by liquidiert** [comment]
Awesome, Thanks! If you need further help with the remaining tests just tell me :)

**Comment by konsultaner** [comment]
@liquidiert Maybe you could write the missing tests according to [codecov](https://codecov.io/gh/konsultaner/connectanum-dart/src/b1dc95dab8eca88d6ea8b9ceb874c0ca7c3bc23a/lib/src/serializer/msgpack/serializer.dart)

**Comment by liquidiert** [comment]
Sure thing 👌

**Pull request merged**